app/routes/connectors/apps/formBuilder.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+
+import { capitalize, lowerCase } from 'lodash';
+import { Controller } from 'react-hook-form';
+
+import { TextField } from '@numaryhq/storybook';
+
+import i18n from '~/src/translations';
+import {
+  ConnectorConfigFormProps,
+  InputType,
+} from '~/src/types/connectorsConfig';
+import { ObjectOf } from '~/src/types/generic';
+
+export const connectorsConfig = {
+  dummypay: {
+    directory: {
+      datatype: 'string',
+      required: true,
+    },
+    fileGenerationPeriod: {
+      datatype: 'duration ns',
+    },
+    filePollingPeriod: {
+      datatype: 'duration ns',
+    },
+  },
+  modulr: {
+    apiKey: {
+      datatype: 'string',
+      required: true,
+    },
+    apiSecret: {
+      datatype: 'string',
+      required: true,
+    },
+    endpoint: {
+      datatype: 'string',
+    },
+  },
+  stripe: {
+    apiKey: {
+      datatype: 'string',
+      required: true,
+    },
+    pollingPeriod: {
+      datatype: 'duration ns',
+    },
+    pageSize: {
+      datatype: 'integer',
+    },
+  },
+  wise: {
+    apiKey: {
+      datatype: 'string',
+      required: true,
+    },
+  },
+};
+
+const inputsFactory = ({
+  inputConfig,
+  name,
+  control,
+  errors,
+  label,
+  parentName,
+}: {
+  inputConfig: InputType;
+  name: string;
+  control: any;
+  errors: ObjectOf<any>;
+  label: string;
+  parentName: string;
+}) => {
+  const textFieldSharedConfig = {
+    label: capitalize(lowerCase(label)),
+    fullWidth: true,
+    required: inputConfig.required,
+    error: !!errors?.[parentName]?.[name],
+    errorMessage:
+      errors?.[parentName]?.[name] &&
+      i18n.t('common.formErrorsMessage.requiredInputs', {
+        inputName: lowerCase(name),
+      }),
+  };
+
+  const controllerSharedConfig = {
+    name: `${parentName}.${name}`,
+    key: name,
+    control: control,
+    rules: { required: inputConfig.required },
+  };
+
```

Comment:
add todo or create methods if you have time
(should be quick)
// TODO create factory methods instead of using case implementation to be more readable